### ISSUE TITLE
In background mode, the pid in background_process_ids is off by one

### ISSUE THREAD

**Issue by 094459** [kaxil/airflowctl#1]
I am trying out airflowctl, and when I setup a new environment and then run it in the background, the PID that is saved in the background_process_ids file is always off by one. It is actually the PID of the shell that ran the command. At the moment I just edit the file, increment the number, and then subsequent commands (stop for eg) then work.

**Comment by kaxil** [comment]
Oh interesting. Which OS are you on?

**Comment by 094459** [comment]
I am on Amazon Linux2 - actually, it is not off by one (sorry, my mistake) I just did it again, and the PID saved is the PID of the shell rather than the actual process.

**Comment by kaxil** [comment]
It should identify the entire process tree and kill them all :( 

Can you share the `ps aux` outputs and contents of `background_process_ids` too please

**Comment by 094459** [comment]
```airflowctl start --background
Verifying Airflow installation...
2.4.3
Airflow is starting in the background (PID: 8743).
Logs are being captured. You can use 'airflowctl logs' to view the logs.
(.venv) ops:~/environment/airflow-jokes $ ps -ef```

```

redacted list

```
ec2-user  8744  0.0  0.0 119864  2204 pts/6    S    17:50   0:00 /bin/sh -c source /home/ec2-user/environment/airflow-jokes/.venv/bin/activate && airflow standalone > /tmp/tmpeo
ec2-user  8745  2.7  1.6 605844 127296 pts/6   Sl   17:50   0:04 /home/ec2-user/environment/airflow-jokes/.venv/bin/python /home/ec2-user/environment/airflow-jokes/.venv/bin/air
ec2-user  8780  4.0  1.3 358520 108324 pts/6   S    17:50   0:06 /home/ec2-user/environment/airflow-jokes/.venv/bin/python /home/ec2-user/environment/airflow-jokes/.venv/bin/air
ec2-user  8782  2.2  1.2 343248 101068 pts/6   S    17:50   0:03 /home/ec2-user/environment/airflow-jokes/.venv/bin/python /home/ec2-user/environment/airflow-jokes/.venv/bin/air
ec2-user  8783  2.8  1.3 426728 102496 pts/6   Sl   17:50   0:04 /home/ec2-user/environment/airflow-jokes/.venv/bin/python /home/ec2-user/environment/airflow-jokes/.venv/bin/air
ec2-user  8809  2.9  1.4 368328 117764 pts/6   S    17:50   0:04 gunicorn: master [airflow-webserver]
ec2-user  8811  0.0  1.2 354820 95876 pts/6    S    17:50   0:00 gunicorn: master [gunicorn]
ec2-user  8812  0.3  1.2 355920 98500 pts/6    S    17:50   0:00 airflow scheduler -- DagFileProcessorManager
ec2-user  8813  0.0  1.2 356904 96104 pts/6    S    17:50   0:00 gunicorn: worker [gunicorn]
ec2-user  8814  0.0  1.2 356904 96104 pts/6    S    17:50   0:00 gunicorn: worker [gunicorn]
ec2-user  8840  0.0  1.3 370412 109540 pts/6   S    17:50   0:00 [ready] gunicorn: worker [airflow-webserver]
ec2-user  8841  0.0  1.3 370412 109476 pts/6   S    17:50   0:00 [ready] gunicorn: worker [airflow-webserver]
```

and the background_proces_ods file is
```
cat .airflowctl/.background_process_ids 
8743
```

**Comment by kaxil** [comment]
Could you install [`pstree`](https://gist.github.com/leafsummer/94beb97f62926b7e1fc399081f4ac45f) and run `pstree 8743`

**Comment by 094459** [comment]
When I start (airflowctl start --background) the PID in the .background_process_ids is 5891. When I do pstree 5891 I get nothing back. When I do pstree for 5892, I get
```
ops:~/environment/airflow-jokes $ pstree 5891
ops:~/environment/airflow-jokes $ pstree 5892
sh───airflow─┬─airflow─┬─airflow schedul
             │         └─gunicorn: maste───2*[gunicorn: worke]
             ├─airflow───gunicorn: maste───2*[[ready] gunicor]
             ├─airflow───{airflow}
             └─3*[{airflow}]
ops:~/environment/airflow-jokes $ 
```

**Comment by 094459** [comment]
I just updated to 0.1.8 to see if I see the same behaviour, and this version still does this.

**Comment by kaxil** [comment]
Fixed in 0.2.0: 

```
pip install airflowctl==0.2.0
```

**Comment by 094459** [comment]
Great stuff, checked and working 100%
